pkg/controllers/node/readiness.go

```diff
@@ -0,0 +1,39 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package node
+
+import (
+	"github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha3"
+	"github.com/awslabs/karpenter/pkg/utils/node"
+	v1 "k8s.io/api/core/v1"
+)
+
+// Readiness is a tiny reconciler that removes the readiness taint when the node is ready
+type Readiness struct {}
+
+// Reconcile removes the NotReady taint when the node is ready
+func (r *Readiness) Reconcile(n *v1.Node) error {
+	if !node.IsReady(n) {
+		return nil
+	}
+	taints := []v1.Taint{}
+	for _, taint := range n.Spec.Taints {
+		if taint.Key != v1alpha3.NotReadyTaintKey {
```

Comment:
Does adding a NotReadyTaint like this cause any problems with scheduling DaemonSets?